src/generators.mjs

```diff
@@ -43,30 +34,43 @@ const createGenerator = markdownInput => {
    */
   const cachedGenerators = { ast: Promise.resolve(markdownInput) };
 
+  const threadPool = new WorkerPool();
+
   /**
    * Runs the Generator engine with the provided top-level input and the given generator options
    *
    * @param {GeneratorOptions} options The options for the generator runtime
    */
-  const runGenerators = async ({ generators, ...extra }) => {
+  const runGenerators = async ({ generators, threads, ...extra }) => {
     // Note that this method is blocking, and will only execute one generator per-time
     // but it ensures all dependencies are resolved, and that multiple bottom-level generators
     // can reuse the already parsed content from the top-level/dependency generators
     for (const generatorName of generators) {
-      const { dependsOn, generate } = availableGenerators[generatorName];
+      const {
+        dependsOn,
+        generate,
+        parallizable = true,
```

Comment:
What is this flag?

Reply:
It's possible that, in the future, we will have a generator that produces non-serializable data, and therefore cannot be parallelized

Reply:
Let's introduce this when we actually need to use this, otherwise dead/mysterious code.